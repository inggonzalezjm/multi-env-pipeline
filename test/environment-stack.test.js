/* eslint-disable no-new */

const { Stack, Tags } = require("aws-cdk-lib");
const { Template } = require("aws-cdk-lib/assertions");
const { EnvironmentStack } = require("../lib/environment-stack");
const { AppConfigs } = require("./configs/configs");

describe("Environment", () => {
  test("snapshot after refactor", () => {
    const stack = new Stack();

    const envParams = AppConfigs.Environments["TestEnv1"];

    const environmentProps = {
      env: {
        account: AppConfigs.Globals.DeploymentAccount,
        region: AppConfigs.Globals.DeploymentRegion,
      },
      envParams: envParams,
      AppConfigs,
    };

    const environment = new EnvironmentStack(stack, "environment-stack", environmentProps);
    if (AppConfigs.Globals.Tags) {
      for (const t in AppConfigs.Globals.Tags) {
        Tags.of(stack).add(t, AppConfigs.Globals.Tags[t]);
      }
    }
    if (envParams.Tags) {
      for (const t in envParams.Tags) {
        Tags.of(stack).add(t, envParams.Tags[t]);
      }
    }
    Tags.of(stack).add("Environment", `${envParams.Name}`);

    const template = Template.fromStack(environment);
    expect(template).toMatchSnapshot();
  });
});
