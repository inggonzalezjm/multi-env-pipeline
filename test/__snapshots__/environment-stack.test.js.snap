// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Environment snapshot after refactor 1`] = `
Object {
  "Parameters": Object {
    "BootstrapVersion": Object {
      "Default": "/cdk-bootstrap/hnb659fds/version",
      "Description": "Version of the CDK Bootstrap resources in this environment, automatically retrieved from SSM Parameter Store. [cdk:skip]",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
  },
  "Resources": Object {
    "buckettestenv12B11321B": Object {
      "DeletionPolicy": "Delete",
      "Properties": Object {
        "BucketEncryption": Object {
          "ServerSideEncryptionConfiguration": Array [
            Object {
              "ServerSideEncryptionByDefault": Object {
                "SSEAlgorithm": "AES256",
              },
            },
          ],
        },
        "PublicAccessBlockConfiguration": Object {
          "BlockPublicAcls": true,
          "BlockPublicPolicy": true,
          "IgnorePublicAcls": true,
          "RestrictPublicBuckets": true,
        },
        "Tags": Array [
          Object {
            "Key": "AppID",
            "Value": "Test12345",
          },
          Object {
            "Key": "AppName",
            "Value": "TestApp",
          },
          Object {
            "Key": "AppVersion",
            "Value": "1.0.0",
          },
          Object {
            "Key": "CostCenter",
            "Value": "Test-54321",
          },
          Object {
            "Key": "Environment",
            "Value": "TestEnv1",
          },
        ],
      },
      "Type": "AWS::S3::Bucket",
      "UpdateReplacePolicy": "Delete",
    },
    "buckettestenv1PolicyCA126837": Object {
      "Properties": Object {
        "Bucket": Object {
          "Ref": "buckettestenv12B11321B",
        },
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "s3:*",
              "Condition": Object {
                "Bool": Object {
                  "aws:SecureTransport": "false",
                },
              },
              "Effect": "Deny",
              "Principal": Object {
                "AWS": "*",
              },
              "Resource": Array [
                Object {
                  "Fn::GetAtt": Array [
                    "buckettestenv12B11321B",
                    "Arn",
                  ],
                },
                Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      Object {
                        "Fn::GetAtt": Array [
                          "buckettestenv12B11321B",
                          "Arn",
                        ],
                      },
                      "/*",
                    ],
                  ],
                },
              ],
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::S3::BucketPolicy",
    },
  },
  "Rules": Object {
    "CheckBootstrapVersion": Object {
      "Assertions": Array [
        Object {
          "Assert": Object {
            "Fn::Not": Array [
              Object {
                "Fn::Contains": Array [
                  Array [
                    "1",
                    "2",
                    "3",
                    "4",
                    "5",
                  ],
                  Object {
                    "Ref": "BootstrapVersion",
                  },
                ],
              },
            ],
          },
          "AssertDescription": "CDK bootstrap stack version 6 required. Please run 'cdk bootstrap' with a recent version of the CDK CLI.",
        },
      ],
    },
  },
}
`;
